imod/typing/grid.py

```diff
@@ -316,3 +316,23 @@ def get_spatial_dimension_names(grid: xu.UgridDataArray) -> list[str]:
 @typedispatch
 def get_spatial_dimension_names(grid: object) -> list[str]:
     return []
+
+
+@typedispatch
+def get_grid_geometry_hash(grid: xr.DataArray) -> int:
+    result = hash(pickle.dumps(grid["x"].values))
+    result += hash(pickle.dumps(grid["y"].values))
+    return result
+
+
+@typedispatch
+def get_grid_geometry_hash(grid: xu.UgridDataArray) -> int:
+    result = hash(pickle.dumps(grid.ugrid.grid.node_x))
+    result = +hash(pickle.dumps(grid.ugrid.grid.node_y))
+    result = +hash(pickle.dumps(grid.ugrid.grid.node_face_connectivity))
```

Comment:
I'm not sure this the appropriate way of hashing these objects.
How likely is is that we get collisions?

For example, this is true:

```python
assert hash(100) + hash(1) == hash(101)
```
Would it be better to do this instead?

```python
hash_x = hash(pickle.dumps(grid.ugrid.grid.node_x)
hash_y = hash(pickle.dumps(grid.ugrid.grid.node_y)
hash_connectivity = hash(pickle.dumps(grid.ugrid.grid.node_face_connectivity)
return hash((hash_x, hash_y, hash_connectivity))
```
```

Reply:
Also the assignment operator is wrong!

```python
result += hash(...)
# versus
result = +hash(...)
```

The last is just assigning the value of to `result`.

Reply:
done (both for structured and unstructured)